File: LeetCode_POTD/July_2026/July_7.py
"""
Easy - do as stated

TC - O(log10(n)), SC - O(1)
"""

class Solution:
    def sumAndMultiply(self, n: int) -> int:
        if n == 0:
            return 0
        s = str(n)
        sum_non_zero = 0
        res = ""
        for c in s:
            if c != "0":
                res += c
                sum_non_zero += int(c)
        return sum_non_zero * int(res)

File: LeetCode_POTD/July_2026/July_7.java
package Leetcode_POTD.July_2026;

/*
Easy - do as stated

TC - O(log10(n)), SC - O(1)
*/

public class July_7 {
    public long sumAndMultiply(int n) {
        long sum=0;
        long m=0;
        int idx=1;
        while(n>0){
            int r=n%10;
            if(r!=0){
                sum+=r;
                m+=r*idx;
                idx*=10;
            }
            n/=10;
        }
        return sum*m;
    }
}
